pandas/core/strings/accessor.py

```diff
@@ -3099,14 +3109,23 @@ def _str_extract_noexpand(arr, pat, flags=0):
 
     regex = re.compile(pat, flags=flags)
     groups_or_na = _groups_or_na_fun(regex)
-    result_dtype = _result_dtype(arr)
+    result_dtype = _result_dtype(arr._orig)
 
     if regex.groups == 1:
-        result = np.array([groups_or_na(val)[0] for val in arr], dtype=object)
+        arr = arr._data.array
+        mask = isna(arr)
+        result = lib.map_infer_mask(
+            np.asarray(arr), groups_or_na, mask.view(np.uint8), convert=False
+        )
+        # special case since extract on an object array converts None to np.nan
+        # will be handled by _str_map array method
+        if is_object_dtype(result_dtype):
+            result[mask] = np.nan
```

Comment:
this is effectively the _str_map array method.

_str_map is not part of the BaseStringArrayMethods interface so shouldn't call it from here, but maybe no big deal since that is internal.

The _str_map method calls maybe_convert_objects so we need to add an extra keyword to so that we always return object arrays.